### PR TITLE
Improved chat

### PR DESCRIPTION
**Pull request by ArmintasP**
- `Chat.cs` and `Message.cs` now have properties with User entities instead of strings.
- If user has an avatar, it is shown in a chat/messaging section. Otherwise a default profile picture is displayed.
- If and only if a new message is received or a new chat session is created, JavaScript functions replace old div elements with new ones.

Note: `update-database` is necessary & deleting old messages/chat relations from database tables might be required because some properties had their type changed from `string` to `User`.

### CODE REVIEW

File: Karma/Migrations/KarmaContextModelSnapshot.cs [hunk 1]
```diff
@@ -115,8 +114,7 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                     b.Property<bool>("IsSeenByPostUser")
                         .HasColumnType("INTEGER");
 
-                    b.Property<string>("PostUserId")
-                        .IsRequired()
+                    b.Property<string>("PostUserIdId")
```
Comment (PauliusLu) [inline Karma/Migrations/KarmaContextModelSnapshot.cs:117]: Why is Id repeated 2 times?

**Review by PauliusLu — APPROVED** [review]
Everything else seems ok.

File: Karma/Migrations/KarmaContextModelSnapshot.cs (hunk 1 above)
Comment (ArmintasP) [reply Karma/Migrations/KarmaContextModelSnapshot.cs:117]: I forgot `add-migration` automatically adds suffix 'id' to a field with a foreign entity. Thanks for noticing, I will rename `PostUserId` to `PostUser`.

**Review by baikauskaite — APPROVED** [review]
Didn't seem to find any problems, looks good to me.

**Pull request merged**